### PR TITLE
[Sage-517] Sage 3 - Search

### LINKED ISSUE #517: Update Search
## Description

Update component and subcomponents to match latest specs in Sage 3.0.

**Dependencies**:
- [Form fields L1](https://github.com/Kajabi/sage-lib/issues/496)

**Links**
- [Figma specs](https://www.figma.com/file/9Km09NjlZHYWsMP7EGT8tI/WIP-Sage-3-%E2%80%94-Admin-Components?node-id=4195%3A20053)
- [Existing component(s)](https://sage-design-system.kajabi.com/pages/component/search)

### PR DESCRIPTION
**Pull request by anechol**
## Description

- Creates Sage mixin for basic styling for floating labels
- Uses mixin for the Form Input and Form Textarea components
- Updates Search to include new floating label

[Figma File](https://www.figma.com/file/9Km09NjlZHYWsMP7EGT8tI/%5BWIP%5D-Sage-3-%E2%80%94-Admin-Components?node-id=4195%3A19716)

## Screenshots

Before:
Screen Shot 2021-09-16 at 4 07 26 PM

After:

Screen Shot 2021-09-16 at 4 07 34 PM

Screen Shot 2021-09-16 at 4 07 43 PM

### Rails
- View [Rails component](http://localhost:4000/pages/component/search)
- Check that component styling updates are there

### React
- View [Storybook](http://localhost:4100/?path=/docs/sage-search--default)
- Check that component styling updates are there

## Testing in `kajabi-products`

(MEDIUM) Minor styling changes for the Search component. Updates **Form Input** and **Form Textarea** components with new floating label mixin. **Please test these components to ensure their styling is undisturbed.**

## Related

Closes #517

### CODE REVIEW

File: packages/sage-assets/lib/stylesheets/components/_form_textarea.scss [hunk 1]
```diff
@@ -60,7 +55,8 @@ $-textarea-color-default-background: sage-color(white);
   &:valid:not(:placeholder-shown),
   &:required:not(:placeholder-shown) {
     ~ .sage-textarea__label {
-      @include floating-label-textarea-active();
+      // @include floating-label-textarea-active();
```
Comment (FuturaExtraBold) [inline packages/sage-assets/lib/stylesheets/components/_form_textarea.scss:58]: This still necessary?

**Comment by FuturaExtraBold** [comment]
@anechol The 'X' close icon in the search field looks off to me:
image

It looks like you could remove this to fix it:
```css
.sage-search--contained .sage-search__reset-button {
    margin-right: 1rem;
}
```

File: packages/sage-assets/lib/stylesheets/components/_form_textarea.scss (hunk 1 above)
Comment (anechol) [reply packages/sage-assets/lib/stylesheets/components/_form_textarea.scss]: Good catch! Removed.

**Review by FuturaExtraBold — APPROVED** [review]
LGTM!

File: packages/sage-assets/lib/stylesheets/components/_form_input.scss [hunk 2]
```diff
@@ -72,20 +72,13 @@ $-input-text-height: sage-font-size(body);
...
-  }
-
   height: $-input-height;
   padding: $-input-padding-offset sage-spacing(sm);
 
   &:focus:not(:disabled),
   &:active:not(:disabled) {
     ~ .sage-input__label {
-      @include floating-label-active();
+      @include sage-form-field-floating-label();
```
Comment (monicawheeler) [inline packages/sage-assets/lib/stylesheets/components/_form_input.scss:81]: 🙌

**Review by pixelflips — APPROVED** [review]
LGTM! 👍🏼 

One small thing I did notice that doesn't seem related to this update is that the search icon in both search inputs turns blue on focus. I'm assuming we'd only want it to change within the input that actually has focus. Not a blocker and could be addressed in a follow-up.

Screen Shot 2021-09-17 at 9 54 31 AM

**Review by monicawheeler — APPROVED** [review]
Looks good!

**Pull request merged**